### PR TITLE
feat: Expressify str.strip_prefix & suffix

### PR DESCRIPTION
**Pull request by reswqa**

### CODE REVIEW

File: crates/polars-ops/src/chunked_array/strings/namespace.rs [hunk 1]
```diff
@@ -334,6 +334,46 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
...
+                _ => Utf8Chunked::full_null(ca.name(), ca.len()),
+            },
+            _ => binary_elementwise(
+                ca,
+                prefix,
+                |opt_s: Option<&str>, opt_prefix: Option<&str>| match (opt_s, opt_prefix) {
+                    (Some(s), Some(prefix)) => Some(s.strip_prefix(prefix).unwrap_or(s)),
+                    _ => None,
+                },
+            ),
```
[the remark this answers is not in the document]
Comment (orlp) [reply crates/polars-ops/src/chunked_array/strings/namespace.rs:353]: Can you make this closure an actual function and annotate the lifetimes?

File: crates/polars-core/src/chunked_array/ops/arity.rs [hunk 2]
```diff
@@ -17,8 +20,8 @@ where
     T: PolarsDataType,
     U: PolarsDataType,
     V: PolarsDataType,
-    F: for<'a> FnMut(Option<T::Physical<'a>>, Option<U::Physical<'a>>) -> Option<K>,
-    V::Array: ArrayFromIter<Option<K>>,
+    F: for<'a> Helper<Option<T::Physical<'a>>, Option<U::Physical<'a>>, Option<V::Physical<'a>>>,
```
[the remark this answers is not in the document]
Comment (orlp) [reply crates/polars-core/src/chunked_array/ops/arity.rs:23]: We definition shouldn't merge it like this with this helper.

File: crates/polars-ops/src/chunked_array/strings/namespace.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (orlp) [reply crates/polars-ops/src/chunked_array/strings/namespace.rs:353]: Pretty much.

[the remark this answers is not in the document]
Comment (orlp) [reply crates/polars-ops/src/chunked_array/strings/namespace.rs:353]: I will take a look tomorrow.

**Comment by orlp** [comment]
@reswqa Well, that was a deep dive into HRTBs and nasty stuff. Seems like I underestimated the problem. Nevertheless, I still wanted to keep the function generic, so could you look at and pull from the strip branch of my repo? https://github.com/orlp/polars/tree/strip

**Comment by orlp** [comment]
@reswqa It only gets inferred wrong if there are two different lifetimes in the input. I'm not happy with it either but I don't know how often that will occur.

**Comment by orlp** [comment]
@reswqa Regarding those errors, that's just that it can't figure out what it should collect into since it immediately pipes into `into_series`. The old code tried to fix that by passing the return type in the signature, what you can do now is this:


```rust
let arr: Float64Chunked = arity::binary_elementwise(lhs, rhs, |opt_l, opt_r| match (
    opt_l, opt_r,
) {
    (Some(l), Some(r)) => {
        if r.is_zero() {
            None
        } else {
            Some(l / r)
        }
    },
    _ => None,
});
Ok(arr.into_series())
```

**Review by ritchie46 — APPROVED** [review]
The rest looks great! Just for my understanding which lifetimes were problematic in the old implementation?

**Comment by orlp** [comment]
@ritchie46 The old code specified the return type of the closure as a parameter `R` on `binary_elementwise`. So then we had (simplifying a bit to explain the problem more clearly):
```rust
F: for<'a> FnMut(&'a str, &'a str) -> R
```
The problem here is that `R` is a type completely independent from the lifetime `'a`. It requires the closure returns the same type `R` for all possible `'a`. But the closure in question wants to return a `&'a str`, which depends on `'a`.

**Comment by ritchie46** [comment]
I see. 👍

**Pull request merged**